master/cmd/determined-master/root.go

```diff
@@ -126,7 +126,7 @@ func mergeCustomSpecs(
 	config map[string]interface{},
 	cp map[string]interface{},
 ) map[string]interface{} {
-	if conf, ok := cp["task_container_defaults"]; ok {
+	if conf, ok := cp["task_container_defaults"].(interface{}); ok {
```

Comment:
should the assertion be for map[string]interface{}? We assert that conf is ```map[string]interface{}``` so this requires that anyway?

Is type asserting for (interface{}) just equivalent to a nil check in this case?

Reply:
asserting that cp is `map[string]interface{}` gives me: `conf (variable of type map[string]interface{}) is not an interface`
so I think if we cast cp into type interface we can cast its result into another type in lines 130 & 133.

Reply:
Can we just remove the two type assertions on line 130 and 133 since we already know the type from 129?

Reply:
No, we still have to define the cpu/gpu pod spec through type asserting conf as a map[string]interface{} once we're sure it exists.